src/logic/queryParams.js

```diff
@@ -35,7 +36,22 @@ import {
  * @return {string}
  */
 const getSearchQueryFromQueryParameterOrLocalStorage = route => {
-  const { [CommonQueryParameters.SearchFilters]: sq } = route?.query;
+  const {
+    [CommonQueryParameters.SearchFilters]: sq,
+    [CommonQueryParameters.PlainSearchFilters]: f,
+  } = route?.query;
+
+  // parse json then serialize
+  if (typeof f === 'string') {
+    console.info('getSearchQueryFromQueryParameterOrLocalStorage', f);
+    try {
+      return serializeFilters(JSON.parse(f));
+    } catch (err) {
+      console.error(err);
+      return '';
```

Comment:
If `f` exists but cannot be parsed by JSON parser or does contains filters but at the same time correct `sq` parameter exists, it will never be returned. It would be better to continue and return `sq` content.

Reply:
I'll remove `return` then ;)

Reply:
I'd make it a `console.warn` and instead of passing `err` print a message explaining what happened. Printing this error will always show the same message: `Uncaught SyntaxError: Unexpected end of JSON input` which does not provide any clues of what happened.

Reply:
ok